### PR TITLE
UseInferredMemberName: only field initializers in anonymous object creation

### PR DESCRIPTION
**Pull request by jcouv**
…should trigger

### Customer scenario
The VB InferredMemberNames analyzer should not trigger on field initializers in object creation, only in anonymous object creation.

```
Dim f = New Foo With {
    .Bar = bar ' <-- Incorrect suggestion to use inferred member name
}
```

### Bugs this fixes
Fixes https://github.com/dotnet/roslyn/issues/23659

### Workarounds, if any
Don't invoke the fixer.

### Performance impact
Low. The fix is localized to the VB InferredMemberName analyzer. Just adding a check.

### Is this a regression from a previous update?
No.

### How was the bug found?
Reported by customers.

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-ide for review. Thanks

**Comment by jcouv** [comment]
@dotnet/roslyn-ide for review. Tiny fix. Thanks

**Comment by jinujoseph** [comment]
@Pilchie for ask mode approval

File: src/EditorFeatures/VisualBasicTest/UseInferredMemberName/UseInferredMemberNameTests.vb [hunk 1]
```diff
@@ -39,6 +39,21 @@ End Class
 ", parseOptions:=s_parseOptions)
         End Function
 
+        <Fact>
+        <WorkItem(23659, "https://github.com/dotnet/roslyn/issues/23659")>
+        Public Async Function TestBug() As Task
```
Comment (Pilchie) [inline src/EditorFeatures/VisualBasicTest/UseInferredMemberName/UseInferredMemberNameTests.vb]: Nit: Test names like this are the worst - `TestMissingForObjectCreation` would be much better.

Comment (jcouv) [reply src/EditorFeatures/VisualBasicTest/UseInferredMemberName/UseInferredMemberNameTests.vb]: My bad. Intended to fix, but forgot.

**Pull request merged**